proto/backend.proto

```diff
@@ -0,0 +1,217 @@
+syntax = "proto3";
+package grafana.plugin;
+
+//-----------------------------------------------
+// Common
+//-----------------------------------------------
+
+message PluginConfig {
+  int64 id = 1;
+  int64 orgId = 2;
+  string name = 3;
+  string type = 4;
+  string url = 5;
+  string jsonData = 6;
+  map<string,string> decryptedSecureJsonData = 7;
+}
+
+//-----------------------------------------------
+// Plugin Status Check
+//-----------------------------------------------
+
+message PluginStatusRequest {
+  // Plugin Configuration
+  PluginConfig config = 1;
+
+  // Environment info
+  map<string,string> headers = 2;
+
+  enum FetchInfo {
+    STATUS = 0;  // nothing
+    API = 1;     // OpenAPI description
+    METRICS = 2; // Prometheus style metrics
+    DEBUG = 3;   // JSON Debug Info (admin+ view)
+  }
+
+  // Get an OpenAPI description (viewer+)
+  FetchInfo fetch = 3;
+}
+
+message PluginStatusResponse {
+  enum PluginStatus {
+    UNKNOWN = 0; // starting etc
+    OK = 1;
+    ERROR = 2;
+  }
+
+  PluginStatus status = 1;
+  string info = 2;
+}
+
+
+//-----------------------------------------------
+// QUERY
+//-----------------------------------------------
+
+message TimeRange {
```

Comment:
I have never really been clear on why both the string and int64 properties exist and what the semantics are.

Reply:
Good question!  I can say what it is and how it is used... but not how it is supposed to be used.

On the fronted, the 'raw' is the input text in the from/to parameters.  It includes things like `now/15m`, and the `xxxMs` version is that parsed to ms.  I think that means that the text verison is most accurate, and the ms verison is just a helper